input_tcp.go

```diff
@@ -66,8 +66,12 @@ func (i *TCPInput) handleConnection(conn net.Conn) {
 	for scanner.Scan() {
 		encodedPayload := scanner.Bytes()
 		// Hex encoding always 2x number of bytes
-		decoded := make([]byte, len(encodedPayload)/2)
-		hex.Decode(decoded, encodedPayload)
+		size := len(encodedPayload)
+		decoded := make([]byte, size/2)
+		_, err := hex.Decode(decoded, encodedPayload[:size-1])
+		if err != nil {
+			log.Fatal("[TCPInput] failed to hex decode TCP payload:", err)
```

Comment:
I'm not sure about `Fatal` since it will stop Gor. Maybe make sense just to print.